cores/nRF5/WInterrupts.c

```diff
@@ -129,14 +168,19 @@ void detachInterrupt(uint32_t pin)
 
   for (int ch = 0; ch < NUMBER_OF_GPIO_TE; ch++) {
     if ((uint32_t)channelMap[ch] == pin) {
+      // Why the memory barrier and NOPs? 
+      // See page 4 at https://web.archive.org/web/20190823222657/http://infocenter.arm.com/help/topic/com.arm.doc.dai0321a/DAI0321A_programming_guide_memory_barriers_for_m_profile.pdf
+      // and recall the AHB runs at 16MHz, so may take 4 cycles before peripheral register is actually updated.
+      NRF_GPIOTE->INTENCLR = (1 << ch);
+      __DSB(); __NOP();__NOP();__NOP();__NOP();
```

Comment:
please remove these _DSB(), the order of these 2 are not important. Plus I don't see the reason why gcc would run it differently.

Reply:
The order is definitely important!  If interrupts are enabled for a device that is not using the correct configuration, it can range from not working, to spurious interrupts, to hard-lock (interrupt storm).

That said, I've removed them ... but advise caution if moving to a different compiler or chip architecture (including multi-core)